Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "API/ToolsApplicationAPI.h"
+#include <AzCore/Settings/SettingsRegistry.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/StringFunc/StringFunc.h>
+#include <AzCore/Utils/Utils.h>
+#include <Prefab/ViewBookmarkLoader.h>
+
+/*!
+ * ViewBookmarkIntereface
+ * Interface for saving/loading View Bookmarks.
+ */
+namespace AzToolsFramework
+{
+    namespace Prefab
+    {
+        static constexpr const char* s_viewBookmarksRegistryPath = "/O3DE/ViewBookmarks/";
+
+        void ViewBookmarkLoader::RegisterViewBookmarkLoaderInterface()
+        {
+            AZ::Interface<ViewBookmarkLoaderInterface>::Register(this);
+        }
+
+        void ViewBookmarkLoader::UnregisterViewBookmarkLoaderInterface()
+        {
+            AZ::Interface<ViewBookmarkLoaderInterface>::Unregister(this);
+        }
+
+        void ViewBookmarkLoader::SaveBookmarkSettingsFile()
+        {
+            auto registry = AZ::SettingsRegistry::Get();
+            if (registry == nullptr)
+            {
+                AZ_Warning("SEditorSettings", false, "Unable to access global settings registry. Editor Preferences cannot be saved");
+                return;
+            }
+
+            // Resolve path to user folder
+            AZ::IO::FixedMaxPath editorBookmarkFilePath =
+                AZ::IO::FixedMaxPath(AZ::Utils::GetProjectPath()) / "user/Registry/ViewBookmarks/";
+
+            editorBookmarkFilePath /= m_bookmarkfileName;
+
+            AZ::SettingsRegistryMergeUtils::DumperSettings dumperSettings;
+            dumperSettings.m_prettifyOutput = true;
+
+            AZStd::string bookmarkKey = "/" + m_bookmarkfileName;
+            dumperSettings.m_includeFilter = [&bookmarkKey](AZStd::string_view path)
+            {
+                AZStd::string_view o3dePrefixPath(bookmarkKey);
+                return o3dePrefixPath.starts_with(path.substr(0, o3dePrefixPath.size()));
+            };
+
+            AZStd::string stringBuffer;
+            AZ::IO::ByteContainerStream stringStream(&stringBuffer);
+            if (!AZ::SettingsRegistryMergeUtils::DumpSettingsRegistryToStream(*registry, "", stringStream, dumperSettings))
+            {
+                AZ_Warning(
+                    "SEditorSettings", false, R"(Unable to save changes to the Editor Preferences registry file at "%s"\n)",
```

Comment:
This should be updated to use the Window paramter value of "ViewBookmark" and to indicate that the file that was unable to be saved was the Bookmark registry file